addons/payment/models/payment_acquirer.py

```diff
@@ -496,7 +496,7 @@ def render(self, reference, amount, currency_id, partner_id=False, values=None):
             values = method(values)
 
         values.update({
-            'tx_url': self._context.get('tx_url', self.get_form_action_url()),
+            'tx_url': self._get_tx_url(values),
```

Comment:
If I understand correctly, you extract this method to be able to generate one-shot payment links based on the payment `values`. Wouldn't it be simpler, and yet compliant with our stable policy, to make it so `values` is now passed as a kwarg to `get_form_action_url`, then use them to generate the payment link in an override of the method?